Sources/TuistKit/Services/DocService.swift

```diff
@@ -0,0 +1,122 @@
+import Foundation
+import RxBlocking
+import Signals
+import TSCBasic
+import TuistCore
+import TuistDoc
+import TuistSupport
+
+// MARK: - DocServicing
+
+protocol DocServicing {
+    func run(path: AbsolutePath) throws
+}
+
+// MARK: - DocService
+
+struct DocService {
+    private let projectGenerator: ProjectGenerating
+    private let swiftDocController: SwiftDocControlling
+    private let swiftDocServer: SwiftDocServing
+    private let fileHandler: FileHandling
+
+    init(projectGenerator: ProjectGenerating = ProjectGenerator(),
+         swiftDocController: SwiftDocControlling = SwiftDocController(),
+         swiftDocServer: SwiftDocServing = SwiftDocServer(),
+         fileHandler: FileHandling = FileHandler())
+    {
+        self.projectGenerator = projectGenerator
+        self.swiftDocController = swiftDocController
+        self.swiftDocServer = swiftDocServer
+        self.fileHandler = fileHandler
+    }
+
+    func run(project path: AbsolutePath, target targetName: String, serve: Bool, port: UInt16) throws {
+        let (_, graph, _) = try projectGenerator.loadProject(path: path)
+
+        let sources = graph.targets(at: path)
+            .filter { !$0.dependsOnXCTest }
+            .flatMap { $0.target.sources }
+            .map { $0.path }
+
+        try withTemporaryDirectory { generationDirectory in
+            let parameters = Parameters(serve: serve,
+                                        swiftDocServer: swiftDocServer,
+                                        port: port,
+                                        generationDirectory: generationDirectory.pathString)
+
+            try swiftDocController.generate(
+                format: parameters.format,
+                moduleName: targetName,
```

Comment:
Q: how does `swift-doc` cope with a `targetName` which does not exist?

Reply:
As far as I've tested it, it's only for the title of the documentation that is generated. You can pass any name you want.